src/__tests__/auto-cleanup.test.tsx

```diff
@@ -1,11 +1,10 @@
-// @flow
 import * as React from 'react';
 import { View } from 'react-native';
 import { render } from '..';
 
 let isMounted = false;
 
-class Test extends React.Component<*> {
+class Test extends React.Component<{ onUnmount?: () => void }> {
```

Comment:
Here you can also do [the same](https://github.com/callstack/react-native-testing-library/pull/881/files#r792167435) treatment `onUnmount?(): void` or keep it as-is